controllers/controllers.go

```diff
@@ -182,3 +174,16 @@ func windowsNodePredicate(byoh bool) predicate.Funcs {
 	}
 
 }
+
+// isValidWindowsNode returns true if the node object has the Windows label and the BYOH
+// label present for only the BYOH nodes based on the value of the byoh boolean parameter.
+func isValidWindowsNode(o client.Object, byoh bool) bool {
```

Comment:
Should `isValidWindowsNode()` helper be unit tested?

Reply:
prefer to keep it simple in this PR given it is a bug fix, adding a unit test is difficult since the input parameter is a reconciled object.

Reply:
Makes sense, thanks @mansikulkarni96